src/sage/libs/gap/gap_test.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+from sage.libs.gap.libgap import libgap
+
+
+@pytest.fixture
+def tmpfile():
+    r"""
+    Temporary file fixture that can be reopened/closed and still
+    clean itself up afterwards.
+    """
+    from tempfile import NamedTemporaryFile
+    from os import unlink
+    t = NamedTemporaryFile(delete=False)
+    yield t
+    unlink(t.name)
+
+
+def test_libgap_can_read_and_write_files(tmpfile):
+    """
+    Test that libgap can write to a temporary file and
+    subsequently read it.
+
+    See :issue:`16502`, :issue:`15833`.
+    """
+    message = "Ceci n'est pas une groupe"
+    libgap.PrintTo(tmpfile.name, message)
+    with open(tmpfile.name) as f:
+        contents = f.read()
+    assert contents == message
+    SystemFile = libgap.function_factory('StringFile')
+    assert SystemFile(tmpfile.name) == libgap(message)
+
+
+def test_gc_loop_1():
+    r"""
+    First stress test for garbage collection in libgap.
```

Comment:
I feel "[[first]] stress test" is highly redundant, mostly because there shouldn't be any need to modify the docstring if we want to reorder the tests or add something inbetween.

Reply:
I deleted "First", "Second", and "Third" from the docstrings. There's still a number in the function name, but I don't know what else to call them.